lib/src/main/java/com/diffplug/spotless/java/GoogleJavaFormatStep.java

```diff
@@ -144,120 +123,24 @@ static final class State implements Serializable {
-			Function<String, String> removeUnused = constructRemoveUnusedFunction(classLoader);
-			return JVM_SUPPORT.suggestLaterVersionOnError(version, (input -> fixWindowsBug(removeUnused.apply(input), version)));
-		}
-
-		private static Function<String, String> constructRemoveUnusedFunction(ClassLoader classLoader)
-				throws NoSuchMethodException, ClassNotFoundException {
-			Class<?> removeUnusedClass = classLoader.loadClass(REMOVE_UNUSED_CLASS);
-			Class<?> removeJavadocOnlyClass;
-			try {
-				// google-java-format 1.7 or lower
-				removeJavadocOnlyClass = classLoader.loadClass(REMOVE_UNUSED_IMPORT_JavadocOnlyImports);
-			} catch (ClassNotFoundException e) {
-				// google-java-format 1.8+
-				removeJavadocOnlyClass = null;
-			}
+			Class<?> formatterFunc = classLoader.loadClass("com.diffplug.spotless.glue.java.GoogleJavaFormatRemoveUnusedImporterFormatterFunc");
+			Constructor<?> constructor = formatterFunc.getConstructor(String.class); //version
+			FormatterFunc googleJavaFormatRemoveUnusedImporterFormatterFunc = (FormatterFunc) constructor.newInstance(version);
 
-			Function<String, String> removeUnused;
-			if (removeJavadocOnlyClass != null) {
-				@SuppressWarnings({"unchecked", "rawtypes"})
-				Object removeJavadocConstant = Enum.valueOf((Class<Enum>) removeJavadocOnlyClass, REMOVE_UNUSED_IMPORT_JavadocOnlyImports_Keep);
-				Method removeUnusedMethod = removeUnusedClass.getMethod(REMOVE_UNUSED_METHOD, String.class, removeJavadocOnlyClass);
-				removeUnused = (x) -> (String) removeUnusedMethod.invoke(null, x, removeJavadocConstant);
-			} else {
-				Method removeUnusedMethod = removeUnusedClass.getMethod(REMOVE_UNUSED_METHOD, String.class);
-				removeUnused = (x) -> (String) removeUnusedMethod.invoke(null, x);
-			}
-			return removeUnused;
+			return JVM_SUPPORT.suggestLaterVersionOnError(version, googleJavaFormatRemoveUnusedImporterFormatterFunc);
 		}
 
-		private static BiFunction<String, Object, String> constructReflowLongStringsFunction(ClassLoader classLoader, Class<?> formatterClazz) throws NoSuchMethodException {
-			Class<?> stringWrapperClass;
-			try {
-				stringWrapperClass = classLoader.loadClass(STRING_WRAPPER_CLASS);
-			} catch (ClassNotFoundException e) {
-				// google-java-format 1.7 or lower, which happen to be LATEST_VERSION_JRE_8, so rely on suggestJre11 for the error
-				return (s, f) -> {
-					throw e;
-				};
-			}
-			Method stringWrapperMethod = stringWrapperClass.getMethod(STRING_WRAPPER_METHOD, String.class, formatterClazz);
-			return (s, f) -> (String) stringWrapperMethod.invoke(null, s, f);
-		}
 	}
 
-	private static final boolean IS_WINDOWS = LineEnding.PLATFORM_NATIVE.str().equals("\r\n");
-
-	/**
-	 * google-java-format-1.1's removeUnusedImports does *wacky* stuff on Windows.
-	 * The beauty of normalizing all line endings to unix!
-	 */
-	static String fixWindowsBug(String input, String version) {
-		if (IS_WINDOWS && version.equals("1.1")) {
```

Comment:
Not that `fixWindowsBug` only runs with GJF 1.1

Reply:
Thanks for pointing this out, I'd totally missed the version guard there! I have removed the `GoogleJavaFormatUtils` and reverted the changes for the testGlue in `lib/build.gradle` as this is now no longer needed.